torchmetrics/image/kid.py

```diff
@@ -0,0 +1,284 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torchmetrics.image.fid import NoTrainInceptionV3
+from torchmetrics.metric import Metric
+from torchmetrics.utilities import rank_zero_warn
+from torchmetrics.utilities.imports import _TORCH_FIDELITY_AVAILABLE
+
+
+def maximum_mean_discrepancy(k_xx: Tensor, k_xy: Tensor, k_yy: Tensor) -> Tensor:
+    """
+    Adapted from https://github.com/toshas/torch-fidelity/blob/v0.3.0/torch_fidelity/metric_kid.py
+    """
+    m = k_xx.shape[0]
+
+    diag_x = torch.diag(k_xx)
+    diag_y = torch.diag(k_yy)
+
+    kt_xx_sums = k_xx.sum(dim=-1) - diag_x
+    kt_yy_sums = k_yy.sum(dim=-1) - diag_y
+    k_xy_sums = k_xy.sum(dim=0)
+
+    kt_xx_sum = kt_xx_sums.sum()
+    kt_yy_sum = kt_yy_sums.sum()
+    k_xy_sum = k_xy_sums.sum()
+
+    value = (kt_xx_sum + kt_yy_sum) / (m * (m - 1))
+    value -= 2 * k_xy_sum / (m * m)
```

Comment:
```suggestion
    value -= 2 * k_xy_sum / (m ** 2)
```

Reply:
@Borda some time ago I saw some benchmarks that multiplication is usually faster for small powers then the actual power operation... Don't remember where though